### ISSUE TITLE
1.12.2 Crash when trying to load into world

### ISSUE THREAD

**Issue by yanphi** [OreCruncher/DynamicSurroundings#436]
*Mod Version:*
DynamicSurroundings-1.12.2-3.4.10.5.jar
*Forge Version:*
forge-14.23.5.2768

*Link to crash log (if applicable):*
https://paste.dimdev.org/uyuyosehus.mccrash (Single Player)
https://paste.dimdev.org/ukavusayed.mccrash (Multiplayer)
*Description:*
Not sure if there's any difference between multiplayer or singleplayer. But I'm crashing ever time I try to load into the world. I had experienced the problem before, but after restarting the game and transferring the file from my server over to SP, the problem had subsequently stopped. But after login in a while later, the problem persists and what I had tried earlier stopped working.

**Comment by OreCruncher** [comment]
Can you update to at least Forge 2779 and see if the problem persists?

**Comment by yanphi** [comment]
Just checked, the file doesn't seem to be there. 

I'm playing from a modpack and when I was looking, I saw that there were also a few people who have encountered the problem as well.

If you scroll down and look in the comments, a few people have posted a link to the crash report, but it isn't detailed with version numbers and such.
https://www.curseforge.com/minecraft/modpacks/roguelike-adventures-and-dungeons

**Comment by OreCruncher** [comment]
I vaguely recall a client side dependency issue that I fixed around that time frame.  Unfortunately I am not sure what version of DS they were referring to.  I loaded up the 3.4.10.5 version in my test environment and did not have any crashes that fit that description.

Would you have a client log from the time of the crash?  I want to look at the init process to see if anything strange is going on.

**Comment by agentwolf88** [comment]
Its only a client side crash. Server is working perfectly.

**Comment by OreCruncher** [comment]
Then I am going to need a client side log.  Usually "class not found" means that an ASM transform blew a gasket and most of the time I can find evidence in the log.

**Comment by agentwolf88** [comment]
https://paste.dimdev.org/hufofadeti.mccrash
This is a client side log.

**Comment by OreCruncher** [comment]
@agentwolf88 That is the same error as 434.  This one is different.  It is complaining about the ClientRegistry class not being found.

**Comment by OreCruncher** [comment]
@agentwolf88 Can you post a link to your client log to issue #434?

**Comment by agentwolf88** [comment]
the log is too large. where can i upload the file?

**Comment by TheSnowyChickens** [comment]
Google Drive is a good try. :=)

**Comment by OreCruncher** [comment]
And ZIP it, too.  Text that is compressed is pretty small.

**Comment by agentwolf88** [comment]
https://drive.google.com/file/d/15yDvxfUSvhJoIwgVpfENwitJzcNfe8nJ/view?usp=sharing

i hope this is the right log

**Comment by OreCruncher** [comment]
OK.  Here is the real culprit:
```
Caused by: java.lang.IllegalArgumentException: Cannot add types to unregistered biome net.minecraft.world.biome.BiomeMushroomIsland@1c66cd02
    at com.google.common.base.Preconditions.checkArgument(Preconditions.java:191)
    at net.minecraftforge.common.BiomeDictionary.addTypes(BiomeDictionary.java:185)
    at net.minecraftforge.common.BiomeDictionary.makeBestGuess(BiomeDictionary.java:285)
    at net.minecraftforge.common.BiomeDictionary.ensureHasTypes(BiomeDictionary.java:382)
    at net.minecraftforge.common.BiomeDictionary.getTypes(BiomeDictionary.java:217)
    at org.blockartistry.lib.BiomeUtils.getBiomeTypes(BiomeUtils.java:104)
    at org.blockartistry.DynSurround.registry.BiomeHandler.<init>(BiomeHandler.java:55)
    at org.blockartistry.DynSurround.registry.BiomeRegistry.register(BiomeRegistry.java:97)
    at org.blockartistry.DynSurround.registry.BiomeRegistry.lambda$init$0(BiomeRegistry.java:119)
    at org.blockartistry.DynSurround.registry.BiomeRegistry$$Lambda$2083/1289289873.accept(Unknown Source)
    at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:408)
    at org.blockartistry.DynSurround.registry.BiomeRegistry.init(BiomeRegistry.java:119)
    at org.blockartistry.DynSurround.registry.RegistryManager$$Lambda$2082/256369716.accept(Unknown Source)
    at java.util.ArrayList.forEach(ArrayList.java:1249)
    at org.blockartistry.DynSurround.registry.RegistryManager.reload(RegistryManager.java:155)
    at org.blockartistry.DynSurround.client.ClientRegistry.<clinit>(ClientRegistry.java:81)
    ... 20 more
```
And I can tell you right now it's not making sense.  It's complaining that a biome cannot be found in the Forge biome registry, but that is where DS got the biome reference to begin with. :\

**Comment by yanphi** [comment]
They shouldn't be? I've realised I have this problem after logging off the server and when I try to login later (such as now) , it doesn't work.

**Comment by OreCruncher** [comment]
OK.  Think it is related to OTG.  I am patching DS to handle the exception more gracefully.

**Comment by yanphi** [comment]
Thanks

**Comment by OreCruncher** [comment]
It's possible.  The reason that it may be OTG related is because of how it handles biome registrations.  Normally for Vanilla biomes are registered when starting up Minecraft.  Because of how OTG works it needs to dynamically change biome when a world actually loads.  This can throw things off with Dynamic Surroundings.  I have patched in code over the years to address some of the side effects of OTG biome behavior - this appears to be another.

**Comment by OreCruncher** [comment]
Fixes pushed: v3.4.10.6 and v3.5.4.2 on CurseForge.
